client/apps/edit/components/content/section_container/index.jsx

```diff
@@ -5,64 +5,69 @@ import { clone, extend, findIndex, findLastIndex } from 'lodash'
 import colors from '@artsy/reaction/dist/Assets/Colors'
 import { IconDrag } from '@artsy/reaction/dist/Components/Publishing'
 import { RemoveButton } from 'client/components/remove_button'
+import { newSection, onChangeSection, removeSection } from 'client/actions/editActions'
 
+import SectionImages from '../sections/images'
 import SectionSlideshow from '../sections/slideshow'
-import { SectionText } from '../sections/text/index.jsx'
+import { SectionText } from '../sections/text'
+import SectionVideo from '../sections/video'
 import { ErrorBoundary } from 'client/components/error/error_boundary'
 import { SectionEmbed } from '../sections/embed'
-import { SectionImages } from '../sections/images'
-import { SectionVideo } from '../sections/video'
 
 export class SectionContainer extends Component {
   static propTypes = {
     article: PropTypes.object.isRequired,
     channel: PropTypes.object.isRequired,
     editing: PropTypes.bool,
-    index: PropTypes.number.isRequired,
+    index: PropTypes.number,
     isHero: PropTypes.bool,
+    newSectionAction: PropTypes.func,
     onRemoveHero: PropTypes.func,
-    onSetEditing: PropTypes.func.isRequired,
-    section: PropTypes.object.isRequired,
-    sections: PropTypes.object
+    onChangeSectionAction: PropTypes.func,
+    onSetEditing: PropTypes.func,
+    removeSectionAction: PropTypes.func,
+    section: PropTypes.object,
+    sections: PropTypes.array
   }
 
   onSetEditing = () => {
     const {
       editing,
       index,
+      isHero,
       onSetEditing
     } = this.props
 
-    const setEditing = editing ? null : index
+    let setEditing
+
+    if (isHero) {
+      setEditing = !editing
+    } else {
+      setEditing = editing ? null : index
+    }
```

Comment:
This block could use a little comment since it takes multiple kinds of data and thus isn't too clear